### ISSUE TITLE
Unittest build fails to compile

### ISSUE THREAD

**Issue by mridsole** [GodotNativeTools/godot-d#63]
On master (78c50b47427fc76bc9443c5218f4bdf2ff95fdaa) with `dmd 2.091.0` (same with `ldc2 1.20`).

```
$ DFLAGS='-unittest' dub build
Performing "$DFLAGS" build using /usr/bin/dmd for x86_64.
godot-d ~master: building configuration "library"...
src/godot/core/variant.d(210,12): Error: mutable method godot.object.GodotObject.baseCasts!().as!(const(Camera)).as is not callable using a const object
src/godot/core/variant.d(210,12):        Consider adding const or inout to godot.object.GodotObject.baseCasts!().as!(const(Camera)).as
src/godot/core/variant.d(276,40): Error: template instance godot.core.variant.Variant.convertsFromGodot!(const(Camera)) error instantiating
src/godot/core/variant.d(318,17):        instantiated from here: compatibleFromGodot!(const(Camera))
src/godot/core/variant.d(318,3):        while evaluating: static assert(compatibleFromGodot!(const(Camera)))
/usr/bin/dmd failed with exit code 1.
```

Seems like it is due to [this `compatibleFromGodot!(const(Camera))` instantiation](https://github.com/GodotNativeTools/godot-d/blob/master/src/godot/core/variant.d#L318).

**Comment by sheepandshepherd** [comment]
Ah, looks like I need to modify the casting templates to accept const objects. Thanks for reporting.

Side note: those checks shouldn't be in a `unittest` block, I should move them somewhere else. D's unit tests won't work with Godot anyway, since these Godot types are only wrappers around C++ types loaded from Godot at run time.

**Comment by sheepandshepherd** [comment]
Const casts are fixed in 00b6710.

For unit-testing Godot code, I need to look into whether the existing frameworks ([GUT](https://github.com/bitwes/Gut) or [WAT](https://github.com/CodeDarigan/WAT)) work with non-GDScript classes. If they do, they might also make it easier to test the language binding automatically in CI.
